### PR TITLE
Fix broken sphinx references.

### LINKED ISSUE #1094: Re-enable sphinx warnings for any references
## Description

Remove the warning suppression for any reference: https://github.com/glotzerlab/hoomd-blue/blob/92dc22d3a253f2c67404bb350420cb8fb1331d5c/sphinx-doc/conf.py#L19

Fix any broken references that result.

## Motivation and context

This warning was suppressed because we kept old v2 API docstrings during the v3 transition. Now that the transition is nearly complete, we have more valid references than invalid reference.

### PR DESCRIPTION
**Pull request by joaander**
## Description

Enable sphinx warnings when references targets are not found and fix all broken references.

## Motivation and context

All sphinx references should point to a valid target.

Resolves #1094 

## How has this been tested?

I build the sphinx documentation locally.

## Change log

```
Changed:

Improved documentation.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/state.py [hunk 1]
```diff
@@ -162,10 +162,10 @@ class State:
...
     respectively. Bonds specify the toplogy used when computing energies and
     forces in `md.bond`, angles define the same for `md.angle`, dihedrals for
-    `md.dihedral` and impropers for `md.improper`. These collectively implement
-    bonding potentials used in molecular dynamics force fields. Like bonds,
-    special pairs define connections between two particles, but special pairs
-    are intended to adjust the 1-4 pairwise interactions in some molecular
+    `md.dihedral` and impropers for `md.improper`. These collectively
+    implement bonding potentials used in molecular dynamics force fields. Like
+    bonds, special pairs define connections between two particles, but special
+    pairs are intended to adjust the 1-4 pairwise interactions in some molecular
```
Comment (tommy-waltmann) [inline hoomd/state.py:168]: Why were these lines changed? If they were not too long before I don't know why they would be too long now.

**Review by tommy-waltmann — COMMENTED** [review]
Just one small thing before approving.

File: hoomd/state.py (hunk 1 above)
Comment (joaander) [reply hoomd/state.py:168]: I had some issues getting `md.improper` to link and tried a few changes to these lines to fix it. Those changes turned out to not be needed, and I failed to undo the change to the wrapping.

**Review by tommy-waltmann — APPROVED** [review]
